File: Sorting/Quick_Sort.c
#include <stdio.h>

void QuickSort(int[],int,int);
void swap(int*,int*);
void Print(int[],int);

int main()
{
 int num;
 printf("Enter Number Of Elements In Array :\n");
 scanf("%d",&num);
 int arr[num],i;
 printf("Enter Elements One After Another :-\n\n");
 
 for(i=0;i<num;i++)
  scanf("%d",&arr[i]);

 printf("Elements In Unsorted Order :-\n\n");
 for(i=0;i<num;i++)
  printf("%d\n",arr[i]);

 QuickSort(arr,0,num);

 printf("Elements In Sorted Order :-\n\n");

 Print(arr,num);

 return 0;
}

void QuickSort(int arr[], int l, int r)           //arr[l...(r-1)]
{
 if((r-1) - l <=0)                           //Base Case : if there is only one element, then return
  return;

 int small=l+1,large;

 for(large=l+1;large<r;large++)
  {
   if(arr[large] <= arr[l])       //arr[l] is the pivot element
    {
     swap(&arr[large],&arr[small]);
     small++;
   }
  }
 swap(&arr[l],&arr[small-1]);

 QuickSort(arr,l,small-1);          //Recursive steps to sort left and right halves
 QuickSort(arr,small,r);
}

void swap(int *x, int *y)
{
 int temp=*x;
 *x=*y;
 *y=temp;
}

void Print(int arr[],int num)
{
 for(int i=0;i<num;i++)
  printf("%d\n",arr[i]);
}
 

File: Searching and Sorting/Quick_Sort.py
def QuickSort(arr, l, r):
    if(l < r):

        pivot_index = Partition(arr, l, r)

        # Recursively sorting left and right subarrays

        QuickSort(arr, l, pivot_index - 1)
        QuickSort(arr, pivot_index + 1, r)

def Partition(arr, l, r):
    pivot = arr[l]  # Taking first element as pivot

    i = r + 1

    for j in range(r, l, -1):
        if(arr[j] > pivot):
            i -= 1
            arr[i], arr[j] = arr[j], arr[i]

    arr[i - 1], arr[l] = arr[l], arr[i - 1]

    return(i-1)

def main():
    ip = list(map(int, input("Enter Input Array: ").split()))
    QuickSort(ip, 0, len(ip) - 1)
    print("Sorted Array is: ", end = "")
    print(*ip)

if __name__ == "__main__":
    main()